### PR TITLE
Make TS recognize input arg name

### PR DESCRIPTION
**Pull request by tugsbayasgalan**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #73253

This PR allows TS schema_matching to match input arg with self for aten operators. This is because, operators in their functional form have input as paremeter instead of self. 

fixes: https://github.com/pytorch/pytorch/issues/71994

Differential Revision: [D34427556](https://our.internmc.facebook.com/intern/diff/D34427556)

### CODE REVIEW

File: torch/csrc/jit/frontend/schema_matching.cpp [hunk 1]
```diff
@@ -230,10 +230,17 @@ static Value* tryMatchArgument(
 
 c10::optional<size_t> findInputWithName(
     const std::string& name,
-    at::ArrayRef<NamedValue> kwargs) {
+    at::ArrayRef<NamedValue> kwargs,
+    bool is_aten) {
```
Comment (gmagogsfm) [inline torch/csrc/jit/frontend/schema_matching.cpp:234]: Is it necessary to check `is_aten`? What if we simply allow `input` to be an alias for `self` universally? (unless there is another argument that's called input). What do you think?

Comment (eellison) [reply torch/csrc/jit/frontend/schema_matching.cpp:234]: if there's a user defined function with both `input` and `self` defined something weird could happen

Comment (anjali411) [reply torch/csrc/jit/frontend/schema_matching.cpp:234]:
```
def f(x):
            return torch.ceil(self=x)

        inp = torch.randn(10)
        self.checkScript(f, (inp, ))
```

doesn't work on master, so should be ok

Comment (tugsbayasgalan) [reply torch/csrc/jit/frontend/schema_matching.cpp:234]: We can have operators with input as their parameter name in the future. So it is probably better to do it for only aten ops for now? @gmagogsfm

Comment (gmagogsfm) [reply torch/csrc/jit/frontend/schema_matching.cpp:234]: Yeah, we should only allow this aliasing when there is no other argument with `self`, `input` as their names. That should be safe.

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34427556).

**Review by gmagogsfm — APPROVED** [review]
Are there any other cases where the schema defined in `native_functions.yaml` is changed during code generation? Because we'd need to copy that change in TorchScript too. Otherwise, it would always treat the original (unchanged) schema in `native_functions.yaml` as source of truth. @anjali411

**Comment by tugsbayasgalan** [comment]
@tugsbayasgalan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34427556).

**Comment by anjali411** [comment]
> Are there any other cases where the schema defined in `native_functions.yaml` is changed during code generation? Because we'd need to copy that change in TorchScript too. Otherwise, it would always treat the original (unchanged) schema in `native_functions.yaml` as source of truth. @anjali411

I am not sure tbh

**Pull request closed without merging**